dexlib2/src/main/java/org/jf/dexlib2/MethodHandleType.java

```diff
@@ -31,59 +31,62 @@
+    public static final int REF_INVOKE_STATIC = 4;
+    /** A method handle for {@code invokespecial}. */
+    public static final int REF_INVOKE_SPECIAL = 7;
+    /** A method handle for invoking a constructor. */
+    public static final int REF_INVOKE_CONSTRUCTOR = 6;
+    /** A method handle for {@code invokeinterface}. */
+    public static final int REF_INVOKE_INTERFACE = 8;
 
-    private static final Map<String, Integer> methodHandleTypeNames = Maps.newHashMap();
-
-    static {
-        methodHandleTypeNames.put("static-put", STATIC_PUT);
-        methodHandleTypeNames.put("static-get", STATIC_GET);
-        methodHandleTypeNames.put("instance-put", INSTANCE_PUT);
-        methodHandleTypeNames.put("instance-get", INSTANCE_GET);
-        methodHandleTypeNames.put("static-invoke", STATIC_INVOKE);
-        methodHandleTypeNames.put("instance-invoke", INSTANCE_INVOKE);
-    }
+    private static final BiMap<Integer, String> methodHandleTypeNames = new ImmutableBiMap.Builder<Integer, String>()
+            .put(REF_GET_FIELD, "ref-get-field")
+            .put(REF_GET_FIELD_STATIC, "ref-get-field-static")
+            .put(REF_PUT_FIELD, "ref-put-field")
+            .put(REF_PUT_FIELD_STATIC, "ref-put-field-static")
+            .put(REF_INVOKE_VIRTUAL, "ref-invoke-virtual")
+            .put(REF_INVOKE_STATIC, "ref-invoke-static")
+            .put(REF_INVOKE_SPECIAL, "ref-invoke-special")
+            .put(REF_INVOKE_CONSTRUCTOR, "ref-invoke-constructor")
+            .put(REF_INVOKE_INTERFACE, "ref-invoke-interface")
+            .build();
 
     @Nonnull public static String toString(int methodHandleType) {
-        switch (methodHandleType) {
-            case STATIC_PUT:
-                return "static-put";
-            case STATIC_GET:
-                return "static-get";
-            case INSTANCE_PUT:
-                return "instance-put";
-            case INSTANCE_GET:
-                return "instance-get";
-            case STATIC_INVOKE:
-                return "static-invoke";
-            case INSTANCE_INVOKE:
-                return "instance-invoke";
-            default:
-                throw new InvalidMethodHandleTypeException(methodHandleType);
+        String val = methodHandleTypeNames.get(methodHandleType);
+        if (val == null) {
+            throw new InvalidMethodHandleTypeException(methodHandleType);
         }
+        return val;
     }
 
     public static int getMethodHandleType(String methodHandleType) {
-        Integer ret = methodHandleTypeNames.get(methodHandleType);
+        Integer ret = methodHandleTypeNames.inverse().get(methodHandleType);
         if (ret == null) {
             throw new ExceptionWithContext("Invalid method handle type: %s", methodHandleType);
         }
         return ret;
     }
 
     public static class InvalidMethodHandleTypeException extends ExceptionWithContext {
+        private static final long serialVersionUID = -2019141391365188343L;
```

Comment:
Nothing else sets this, so let's leave it out to be consistent. If it becomes a problem for some reason, I would prefer to add these to all of the custom exception/serializable classes in one fell swoop.

Reply:
This I can change. I was probably just trying to get rid of warnings and forgot I did it.